lib/ckb/wallets/anyone_can_pay_wallet.rb

```diff
@@ -0,0 +1,187 @@
+        @sudt_args = sudt_args
+        @sudt_type_script = CKB::Types::Script.new(code_hash: CKB::Config::SUDT_CODE_HASH, args: sudt_args, hash_type: "data")
+        @anyone_can_pay_cell_lock_scripts = (anyone_can_pay_addresses.is_a?(Array) ? anyone_can_pay_addresses : [anyone_can_pay_addresses]).map do |address|
+          script = CKB::AddressParser.new(address).parse.script
+          raise "not an anyone can pay address" if script.code_hash != CKB::Config::ANYONE_CAN_PAY_CODE_HASH
+
+          script
+        end
+        @is_owner = input_scripts.size == 1 && input_scripts.first.compute_hash == anyone_can_pay_cell_lock_scripts.first.compute_hash
+        @need_sudt = true
+      end
+
+      def generate(to_address, transfer_info, output_info ={}, fee_rate = 1)
+        transfer_type, transfer_amount = transfer_info[:type], transfer_info[:amount]
+        type = is_owner ? output_info[:type] : sudt_type_script
+        case transfer_type
+        when :ckb
+          capacity = transfer_amount
+          data = "0x#{'0' * 32}"
+          @need_sudt = false
+        when :udt
+          capacity = is_owner ? CKB::Utils.byte_to_shannon(142) : 0
+          data = CKB::Utils.generate_sudt_amount(transfer_amount)
+        else
+          raise "wrong transfer type, only support ckb or udt"
+        end
+
+        advance_generate(
+          to_infos: {
+            to_address => { capacity: capacity, type: type, data: data }
+          },
+          contexts: [output_info[:context]],
+          fee_rate: fee_rate
+        )
+      end
+
+      def advance_generate(to_infos: , contexts: [], fee_rate: 1)
+        outputs = []
+        outputs_data = []
+
+        to_infos.each do |address, output_info|
+          lock = CKB::AddressParser.new(address).parse.script
+          raise "unexpected anyone can pay address" if lock.code_hash == CKB::Config::ANYONE_CAN_PAY_CODE_HASH && !anyone_can_pay_cell_lock_scripts.map(&:compute_hash).include?(lock.compute_hash)
+
+          capacity = is_owner && output_info[:type] && output_info[:type].compute_hash == sudt_type_script.compute_hash ? [output_info[:capacity], CKB::Utils.byte_to_shannon(142)].max : output_info[:capacity]
+          outputs << CKB::Types::Output.new(capacity: capacity, lock: lock, type: output_info[:type])
+          outputs_data << (output_info[:data] || "0x")
+        end
+
+        if outputs.all? { |output| output.capacity > 0 } || (outputs.map { |output| output.capacity }.sum == 0)
+          if to_infos.any? { |_, output_info| output_info[:type] && output_info[:type].compute_hash == sudt_type_script.compute_hash }
+            outputs << CKB::Types::Output.new(capacity: 0, lock: input_scripts[-1], type: nil)
+            outputs_data << "0x"
+            outputs << CKB::Types::Output.new(capacity: CKB::Utils.byte_to_shannon(142), lock: input_scripts[-1], type: sudt_type_script)
```

Comment:
142 to a const ?

Reply:
changed. https://github.com/nervosnetwork/ckb-sdk-ruby/pull/244/commits/90aa364444e449b2c9b87c6448085856a4d92cad